components/authz-service/server/v2/project_update_manager_test.go

```diff
@@ -125,28 +151,35 @@ func TestProjectUpdateManagerSendCancelEvent(t *testing.T) {
 		true,               // completed
 		event_ids.ComplianceInspecReportProducerID)
 
-	manager.ProcessStatusMessage(complianceStatusEvent)
+	manager.ProcessStatusEvent(complianceStatusEvent)
 
-	assert.Equal(t, v2.NotRunningState, manager.State())
+	waitForWithTimeout(t, func() bool {
+		return config.NotRunningState == manager.State()
+	}, time.Second*3, "State did not switch to NotRunning")
 }
 
 func TestProjectUpdateManagerNoCancelEventSent(t *testing.T) {
-	var lastestPublishedEvent *automate_event.EventMsg
+	var eventsSent []*automate_event.EventMsg
 	mockEventServiceClient := automate_event.NewMockEventServiceClient(gomock.NewController(t))
 	mockEventServiceClient.EXPECT().Publish(gomock.Any(), gomock.Any()).AnyTimes().DoAndReturn(
 		func(cxt interface{}, in *automate_event.PublishRequest) (*automate_event.PublishResponse, error) {
-			lastestPublishedEvent = in.Msg
+			eventsSent = append(eventsSent, in.Msg)
 			return &automate_event.PublishResponse{}, nil
 		})
-	manager := v2.NewProjectUpdateManager(mockEventServiceClient)
-	assert.Equal(t, v2.NotRunningState, manager.State())
+	manager := v2.NewProjectUpdateManager(mockEventServiceClient, config.NewManager(""))
+	assert.Equal(t, config.NotRunningState, manager.State())
 
 	err := manager.Start()
 	assert.NoError(t, err)
-	assert.Equal(t, v2.RunningState, manager.State())
+	waitForWithTimeout(t, func() bool {
+		return config.RunningState == manager.State()
+	}, time.Second*3, "State did not switch to Running")
 
-	eventData := lastestPublishedEvent.Data
+	waitFor(func() bool {
+		return len(eventsSent) > 0
+	})
 
+	eventData := eventsSent[0].Data
```

Comment:
This is just a test, so maybe it doesn't matter, but could `eventsSent` ever be `nil` or empty?

Reply:
I don't think it could be `nil` because the 3 lines before are waiting until the `eventsSent` has at least one value in it.